zerver/lib/markdown/__init__.py

```diff
@@ -2191,12 +2191,12 @@ def build_inlinepatterns(self) -> markdown.util.Registry:
         # Custom strikethrough syntax: ~~foo~~
         DEL_RE = r"(?<!~)(\~\~)([^~\n]+?)(\~\~)(?!~)"
         # Custom bold syntax: **foo** but not __foo__
-        # str inside ** must start and end with a word character
-        # it need for things like "const char *x = (char *)y"
-        EMPHASIS_RE = r"(\*)(?!\s+)([^\*^\n]+)(?<!\s)\*"
-        STRONG_RE = r"(\*\*)([^\n]+?)\2"
-        STRONG_EM_RE = r"(\*\*\*)(?!\s+)([^\*^\n]+)(?<!\s)\*\*\*"
-        TEX_RE = r"\B(?<!\$)\$\$(?P<body>[^\n_$](\\\$|[^$\n])*)\$\$(?!\$)\B"
+        # String inside ** must start and end with a word
+        # character; this makes us CommonMark compliant.
+        EMPHASIS_RE = r"(\*)(?!\s+)([^\*^\n]+?)(?<!\s)\2"
+        STRONG_RE = r"(\*{2})(?!\s+)(([^\*^\n]+?)\*?([^\*^\n]+?))(?<!\s)\2"
```

Comment:
You’ve added two extra unused capturing groups and a ReDoS vulnerability. I’m not sure what you meant to write, but this wasn’t it.

Reply:
@andersk The original regex matches with: `** strong **`, `**strong **`, `** strong**`.
With the modified regex, I tried to make it only match for the cases like: `**strong**`, `**stro*ng**`. (For unused capturing groups, we can make it non-capturing groups)

Reply:
But since this looks complex, we should leave this?